src/foam/nanos/crunch/ui/CapableView.js

```diff
@@ -101,9 +104,11 @@ foam.CLASS({
       .end();
     },
 
-    // add listeners to payloads
-    function addListeners() {
-      for ( payload of this.capableObj.capablePayloads ) {
+    // add listeners to payload data
+    function listenOnPayloads() {
+      for ( const payload of this.capableObj.capablePayloads ) {
+        if ( payload.data === null ) continue;
```

Comment:
Takes a little less mental effort for people to figure out code if you write it as either:
if ( payload.data) payload.data.sub(...).  OR:
payload.data && payload.ata.sub(...)

Reply:
I want to make a clear distinction between when data is set to undefined and when data is set to null because when data is null, it is intended but when data is undefined, it is an error. How about

```
if (data === undefined) raise/notify an error

// else
(data !== null) payload.data.sub(...)
```

Reply:
If it is either null or undefined, you're still going to get a NPE. Is that what you want?

Reply:
@kgrgreer yes In both cases its is NPE and I dont want to add a listener. But when data is undefined, there is some flaw in logic so I want to log it for debugging purpose.

Reply:
How about 

 ```
if ( payload.data ) {
          payload.data.sub(this.clonePayloads);
 } else if ( payload.data === undefined ) {
          console.warn(`${payload} cannot have data as undefined`);
 }
```

Reply:
@kgrgreer Now I think about it, I think the function that sets payload data should deal with undefined data.

Reply:
Addressed top comment